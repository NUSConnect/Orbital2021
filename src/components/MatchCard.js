import React, { useState, useEffect } from 'react'
import * as firebase from 'firebase'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Card } from 'react-native-shadow-cards'

const MatchCard = ({ navigation, route, item, onPress }) => {
  const currentUserId = firebase.auth().currentUser.uid
  const color = item.success ? '#90EE90' : '#FF7F7F'
  const [otherName, setOtherName] = useState('')

  const getOtherName = async () => {
    if (item.isGroup || item.isGroup === undefined) {
      console.log('Group')
    } else {
      const otherId = item.users.filter(x => x !== currentUserId)[0]
      await firebase
        .firestore()
        .collection('users')
        .doc(otherId)
        .get()
        .then(documentSnapshot => setOtherName(documentSnapshot.data().name))
    }
  }

  useEffect(() => {
    getOtherName()
  }, [])

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={() => console.log('pressed')}>
        <Card style={{ padding: 10, margin: 10, backgroundColor: color, height: 120 }}>
          <Text style={styles.title}> {item.success ? 'Success!' : 'Sorry!'} </Text>
          {item.success
            ? item.isGroup
                ? <Text style={styles.info}> Match successful! Tap here to chat with group </Text>
                : <Text style={styles.info}> Match successful! Tap here to chat with {otherName}</Text>
            : <Text style={styles.info}> Match failed, better luck next time! </Text>}
        </Card>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingTop: 5,
    paddingBottom: 5
  },
  title: {
    fontSize: 22,
    color: '#000000',
    fontWeight: '600'
  },
  info: {
    fontSize: 18,
    color: '#000000',
    fontWeight: '600'
  }
})

export default MatchCard
