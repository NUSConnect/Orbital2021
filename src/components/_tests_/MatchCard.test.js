import * as React from 'react'
import MatchCard from '../MatchCard'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.useFakeTimers()

jest.mock('firebase', () => {
  return {
    auth: jest.fn().mockReturnValue({ currentUser: { uid: 22222 } })
  }
})

const mockedNavigate = jest.fn()

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate
    })
  }
})

class CustomDate {
  constructor (date) {
    this.date = date
  }

  toDate () {
    return this.date
  }
}

const setDate = new Date(2021, 5, 15, 12, 0, 0)
const timestamp = new CustomDate(setDate)

const mockItem = {
  success: true,
  timeMatched: timestamp,
  isGroup: true
}

it('renders correctly', () => {
  const tree = renderer.create(<MatchCard item={mockItem} />)
  expect(tree).toMatchSnapshot()
})

it('on press card', () => {
  console.log = jest.fn()
  const { getByTestId } = render(<MatchCard item={mockItem} />)
  fireEvent.press(getByTestId('pressable'))
  expect(console.log).toHaveBeenCalledWith('group or fail')
})
