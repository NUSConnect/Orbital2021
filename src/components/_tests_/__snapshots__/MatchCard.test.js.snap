// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingBottom": 5,
      "paddingTop": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="pressable"
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 5,
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 3,
            "width": 710,
          },
          Object {
            "backgroundColor": "#90EE90",
            "borderRadius": 15,
            "height": 100,
            "margin": 10,
            "padding": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 22,
            "fontWeight": "600",
          }
        }
      >
        Success!
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "300",
          }
        }
      >
        Match successfully completed on 
        Tue Jun 15 2021
        ! Tap here to chat with your new group!
      </Text>
    </View>
  </View>
</View>
`;
